### PR TITLE
Use a github app token, not the default token, to run !merge_upstream workflow actions

### PR DESCRIPTION
**Pull request by S34NW**
## What Does This PR Do
Swaps github's default token for an app token when running certain steps of the `!merge_upstream` workflow.

## Why It's Good For The Game
Currently, running the `!merge_upstream` workflow does not trigger a CI run, which is very annoying, this is due to the following GH policy:
```
When you use the repository's GITHUB_TOKEN to perform tasks, events triggered by the GITHUB_TOKEN,
with the exception of workflow_dispatch and repository_dispatch, will not create a new workflow run.
This prevents you from accidentally creating recursive workflow runs.
For example, if a workflow run pushes code using the repository's GITHUB_TOKEN,
a new workflow will not run even when the repository contains a workflow configured to run when push events occur.
```

This bypasses that on the repository by using an app that (hopefully) has the right repository permissions to carry out the steps in the workflow, without compromising workflow security by giving it broad perms.

## Testing
This is like the most impossible thing to test without a repo and org to use for testing, so we are just gonna do it live I guess.

### CODE REVIEW

**Comment by AffectedArc07** [comment]
Can you also do this for nanomap updates? Those need closing and reopening to trigger CI on them.

File: .github/workflows/render_nanomaps.yml [hunk 1]
```diff
@@ -33,6 +39,8 @@ jobs:
       run: './tools/github-actions/nanomap-renderer-invoker.sh'
 
     - name: 'Commit Maps'
+      env:
+        GH_TOKEN: ${{ steps.create_token.outputs.token }}
```
[the remark this answers is not in the document]
Comment (S34NW) [reply .github/workflows/render_nanomaps.yml:43]: Are you saying it persists across steps once set?

**Review by AffectedArc07 — APPROVED** [review]
I trust this

**Comment by S34NW** [comment]
> I trust this

Mans lost it

**Review by lewcc — APPROVED** [review]

**Pull request merged**